manifests/init.pp

```diff
@@ -1,24 +1,19 @@
-# -- Class systemd  
-# This module allows triggering systemd commands once for all modules 
+# This module allows triggering systemd commands once for all modules
+#
+# @api public
+#
+# @param service_limits
+#   May be passed a resource hash suitable for passing directly into the
+#   ``create_resources()`` function as called on ``systemd::service_limits``
+#
 class systemd (
-  $service_limits = {}
+  Optional[Hash] $service_limits = undef
```

Comment:
Why not make the default `{}` and check for the size of the hash? I find this a bit more logical than allowing for `undef` as a default value.